app/models/notification_group.rb

```diff
@@ -6,8 +6,11 @@ class NotificationGroup < ActiveModelSerializers::Model
   # Try to keep this consistent with `app/javascript/mastodon/models/notification_group.ts`
   SAMPLE_ACCOUNTS_SIZE = 8
 
-  def self.from_notification(notification, max_id: nil)
-    if notification.group_key.present?
+  def self.from_notification(notification, max_id: nil, grouped_types: nil)
+    grouped_types = grouped_types.presence&.map(&:to_sym) || Notification::GROUPABLE_NOTIFICATION_TYPES
```

Comment:
Same question as above: Is this the desired behavior when `grouped_types` are missing?